refactor (aboutuser, closes #169)

sashwatS opened the pull request:
added different privelege levels if its the user or if its someone else on the about user page. i don't know if it works because all I can see atm in that page is a hamster spinning on a wheel and 404s.

Nikolay-Ts reviewed (changes requested):
- need JSDOC documnetation for AboutUser.tsx
- Make a timeout on the use effect. For example if it is trying to load the data for over 5 seconds give some message and reroute back to the "/"

sashwatS commented:
okay, added a mock data file and I added a handler for that in about user. the handler for the real backend handler is still in there but commented out for now. also I needed to change app.tsx and remove protected route because otherwise the user URLs would not render, hence the merge conflict. idk what to do about that so pls advise

sashwatS commented:
resolved the merge conflict; deleted the protected route in dev for aboutuser so that the webpages would properly render. there is probably a better solution for this in the future but for this deliverable I think its okay

Nikolay-Ts reviewed (changes requested):
- about user page needs to be nested within the user/:userId/aboutUser
- update dev and merge it into your branch as you are getting type errors and that will fix it

sashwatS commented:
about user can be accessed by user/id/aboutuser now

Nikolay-Ts reviewed (changes requested):
- overal great
- make the routing camelCase
- add the route to the README
- bug with the use effect, when a incorrect user id is sent, only one window alertt should be given, instead you get spammed, sent to login, spammed again and sent to the landing page

Nikolay-Ts reviewed (changes requested):
- can you make it work using the data being fetched from the DB. 
- if you want we can make that a new PR or continue on this one, your choice

sashwatS commented:
hopefully final fix 🤞
